### PR TITLE
ENT-11871: Upgrade TokenToFriend CorDapp to JDK17

### PR DESCRIPTION
**Pull request by jakubzadroga**
Upgrade the CorDapp to JDK17.

### CODE REVIEW

File: Tokens/tokentofriend/clients/build.gradle [hunk 1]
```diff
@@ -23,6 +23,7 @@ dependencies {
     implementation 'com.google.code.gson:gson:2.8.5'
 
     // Corda dependencies.
+    implementation "$corda_release_group:corda-core:$corda_release_version"
```
Comment (tomstark99) [inline Tokens/tokentofriend/clients/build.gradle:26]: what is the need to add corda-core here?

Comment (jakubzadroga) [reply Tokens/tokentofriend/clients/build.gradle:26]: Something must have changed between 4.11 and 4.12 around corda-core, I had to add explicit dependency for it in quite a few CorDapps. Without it the `Client` won't compile:
image

File: Tokens/tokentofriend/clients/build.gradle [hunk 2]
```diff
@@ -34,6 +35,7 @@ dependencies {
     implementation "org.apache.logging.log4j:log4j-slf4j2-impl:${log4j_version}"
     implementation "org.apache.logging.log4j:log4j-web:${log4j_version}"
     implementation "org.slf4j:jul-to-slf4j:$slf4j_version"
+    implementation "org.bouncycastle:bcprov-jdk18on:1.75"
```
Comment (adelel1) [inline Tokens/tokentofriend/clients/build.gradle:38]: @jakubzadroga This is the wrong version of BC, we upgraded to LTS version.

Comment (jakubzadroga) [reply Tokens/tokentofriend/clients/build.gradle]: done

**Review by adelel1 — APPROVED** [review]

**Pull request merged**